anomali-limo-feed/operations.py

```diff
@@ -0,0 +1,217 @@
+""" 
+Copyright start 
+Copyright (C) 2008 - 2021 Fortinet Inc. 
+All rights reserved. 
+FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE 
+Copyright end 
+"""
+import base64
+
+import requests
+from connectors.cyops_utilities.builtins import create_file_from_string
+from taxii2client.v20 import Collection, as_pages
+
+from connectors.core.connector import get_logger, ConnectorError
+
+logger = get_logger('anomali-limo-feed')
+
+
+class TaxiiClient(object):
+    def __init__(self, config):
+        self.server_url = config.get('server_url')
+        if not self.server_url.startswith('https://'):
+            self.server_url = 'https://' + self.server_url
+        if not self.server_url.endswith('/'):
+            self.server_url += '/'
+        self.username = config.get('username')
+        self.password = config.get('password')
+        self.verify_ssl = config.get('verify_ssl')
+
+    def make_request_taxii(self, endpoint=None, method='GET', data=None, params=None, files=None, headers=None):
+        try:
+            if endpoint:
+                url = self.server_url + 'api/v1/taxii2/feeds/' + endpoint
+            else:
+                url = self.server_url + 'api/v1/taxii2/feeds/'
+            usr_pass = self.username + ":" + self.password
+            usr_pass = usr_pass.encode()
+            b64val = base64.b64encode(usr_pass)
+            token = 'Basic {}'.format(b64val.decode("utf-8"))
+            default_params = {'match[type]': 'indicator'}
+            params = {**default_params, **params} if params is not None and params != '' else default_params
+            default_header = {'Authorization': token, 'Content-Type': 'application/json'}
+            headers = {**default_header, **headers} if headers is not None and headers != '' else default_header
+            response = requests.request(method, url, params=params, files=files, data=data, headers=headers,
+                                        verify=self.verify_ssl)
+            if response.status_code == 200 or response.status_code == 206:
+                return response.json()
+            else:
+                logger.error(response.text)
+                raise ConnectorError({'status_code': response.status_code, 'message': response.reason})
+        except requests.exceptions.SSLError:
+            raise ConnectorError('SSL certificate validation failed')
+        except requests.exceptions.ConnectTimeout:
```

Comment:
there is no timeout specified in the request, hence this error will not be raised. Will be good to add a 2 min timeout